pepper-apis/src/main/java/org/broadinstitute/ddp/route/AdminCreateUserLoginAccountRoute.java

```diff
@@ -0,0 +1,178 @@
+
+        LOG.info("Attempting to create login account for user {} in study {} by operator {}",
+                userGuid, studyGuid, ddpAuth.getOperator());
+
+        response.type(ContentType.APPLICATION_JSON.getMimeType());
+        TransactionWrapper.useTxn(handle -> {
+            StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyGuid);
+            if (studyDto == null) {
+                String msg = "Could not find study with guid " + studyGuid;
+                LOG.warn(msg);
+                throw ResponseUtil.haltError(HttpStatus.SC_NOT_FOUND, new ApiError(ErrorCodes.NOT_FOUND, msg));
+            }
+
+            var userDao = handle.attach(UserDao.class);
+            User user = userDao.findUserByGuid(userGuid).orElse(null);
+            if (user == null) {
+                String msg = "Could not find user with guid " + userGuid;
+                LOG.warn(msg);
+                throw ResponseUtil.haltError(HttpStatus.SC_NOT_FOUND, new ApiError(ErrorCodes.NOT_FOUND, msg));
+            }
+
+            EnrollmentStatusType status = handle.attach(JdbiUserStudyEnrollment.class)
+                    .getEnrollmentStatusByUserAndStudyIds(user.getId(), studyDto.getId())
+                    .orElse(null);
+            if (status == null) {
+                LOG.error("Attempted to create login account for user not in study, userGuid={}, studyGuid={}", userGuid, studyGuid);
+                throw ResponseUtil.haltError(HttpStatus.SC_NOT_FOUND, new ApiError(ErrorCodes.NOT_FOUND, "User not in study"));
+            } else if (status.isExited()) {
+                LOG.error("Attempted to create login account for exited user, userGuid={}, studyGuid={}", userGuid, studyGuid);
+                throw ResponseUtil.haltError(HttpStatus.SC_NOT_FOUND, new ApiError(ErrorCodes.NOT_FOUND, "User not in study"));
+            } else {
+                LOG.info("Found user {} in study {} with status {}", userGuid, studyGuid, status);
+            }
+
+            if (user.hasAuth0Account()) {
+                LOG.error("Attempted to create login account for user with an existing auth0 account, userGuid={}, studyGuid={}",
+                        userGuid, studyGuid);
+                throw ResponseUtil.haltError(HttpStatus.SC_BAD_REQUEST,
+                        new ApiError(ErrorCodes.OPERATION_NOT_ALLOWED, "User already has login account"));
+            } else if (user.isTemporary()) {
+                LOG.error("Attempted to create login account for temporary user, userGuid={}, studyGuid={}", userGuid, studyGuid);
+                throw ResponseUtil.haltError(HttpStatus.SC_BAD_REQUEST,
+                        new ApiError(ErrorCodes.OPERATION_NOT_ALLOWED, "User is a temporary user"));
+            }
+
+            Auth0ManagementClient auth0Mgmt = createManagementClient(handle, studyGuid);
+            String auth0ClientId = payload.getAuth0ClientId();
```

Comment:
Hmm. Should client be dictating what the Auth0 client id? This user is authorized already on this study and and server know what the client id. Maybe we don't need this coming from client?

Reply:
> user is authorized already

By "user" do you mean the study admin? This `auth0ClientId` is different. This is the client to _sign up the user with_. Whenever a user signs up in Auth0, they are associated with an auth0 database connected to the client. If we're using separate clients in `testboston` (e.g. one for CRC and one for regular app), then I imagine we won't want the users to be in the CRC client's database connection, if it even has one. I'm not sure how we'll figure it out in the backend, so leaving it up to the caller.

Reply:
Not understanding how client would have a better idea what auth0 client id to use to sign up a subject on a study than the server. I was expecting that there would only be one valid Auth0 client that could be used to signup subjects. Perhaps this API call is meant to extend to admin signups ? Would expect that in that case too there would only be one valid value. Not seeing it, but maybe I am missing an important detail here.

Reply:
No, this is not meant for admin signups. Perhaps I'm misunderstanding things, but here's some of my notes:

* A study may have multiple clients. E.g. one for Angular, one for mobile, one for admin interface, etc.
* A client _does not_ have to have a database connection, e.g. only social.
* A database connection is required to sign up a user with email/password.
* The client used to call this API (aka the client that authenticated the operator who is calling this API) might not be the one we want to use to sign up users.

Given what I know, I'm not sure how the server will figure out what client to use. Can it be an arbitrary choice? Perhaps we need more study configuration to inform the server. Any help or suggestion here is appreciated.

Also, my intent here is that for `testboston`, the CRC client calling this will pass in the client id of the regular app client because that's the one where other invited users are signing up with. Maybe this is asking too much of frontend clients. I'll figure something else out then.

Reply:
Okay, added some code to infer client in the backend for now. We can make it more flexible by adding study configuration later. The request just takes `email` now.